flake8_black.py

```diff
@@ -63,12 +65,44 @@ def parse_options(cls, options):
         # cls.black_check = bool(options.black)
         cls.line_length = int(options.max_line_length)
         # raise ValueError("Line length %r" % options.max_line_length)
+
+    def _load_black_config(self):
+        source_path = (
+            self.filename
+            if self.filename not in self.STDIN_NAMES
+            else Path.cwd().as_posix()
+        )
+        project_root = black.find_project_root((Path(source_path),))
+        path = project_root / "pyproject.toml"
+
+        if path.is_file():
+            pyproject_toml = toml.load(str(path))
+            config = pyproject_toml.get("tool", {}).get("black", {})
+            return {k.replace("--", "").replace("-", "_"): v for k, v in config.items()}
+        return None
+
+    @property
+    def file_mode(self):
         try:
-            cls.file_mode = black.FileMode(line_length=cls.line_length)
+            black_config = self._load_black_config()
+            if black_config:
+                target_versions = {
+                    black.TargetVersion[val.upper()]
+                    for val in black_config.get("target_version", [])
+                }
+
+                return black.FileMode(
+                    target_versions=target_versions,
+                    line_length=self.line_length or black_config.get("line_length", 88),
```

Comment:
So the flake8 setting for line length is still used in preference to any in the ``pyproject.toml`` file? This should go in the README.

Reply:
@peterjc 
Actually, this is a tricky moment, because if someone defines different values for line-length for flake and black, his project will fails on build or will stay in inconsnstent state.

I think, that if we find this conflict in configurations, we can notify user about this. But I also think, that we should use line-length from .flake8 in this case because black is running as submodule of flake. 

What do you think about notifying but enforcing flake rules?

Reply:
I don't know how we can raise a configuration warning like this from within a flake8 plugin, but agree in principle that an error or warning would be ideal if the two line length settings disagreed.

Reply:
Maybe we could perhaps raise a new ``BLK9##`` error (once) against the first file we are asked to check?

Reply:
@peterjc make sense! I add BLK997, which stands for configuration conflicts. Please check updates in PR